### PR TITLE
Add NHC to Resources

### PR DESCRIPTION
**Pull request by omnivector-kelly**
**Please provide description of the purpose of this pull request, as well as its
motivation and context.**

Is it a new feature? A bug fix? Does it address an existing issue?
This is a feature to add NHC as a resource.

**How was the code tested?**
make tests

Final checklist
- [x] I am the author of these changes, or I have the rights to submit them.
- [x] I added the relevant changed to the README and/or documentation.
- [x] I self reviewed my own code.
- [x] I updated the `CHANGELOG` according to the [contributing
  guide](https://omnivector-solutions.github.io/osd-documentation/master/contributing.html#changelog)

### CODE REVIEW

File: .github/workflows/publish-candidate.yaml [hunk 1]
```diff
@@ -68,6 +68,7 @@ jobs:
           release="--release=edge --release=candidate"
 
           charmcraft upload $release slurmctld$extension --resource etcd:1
+          charmcraft upload $release slurmd$extension --resource nhc:1
 
           for charm in slurmd slurmdbd slurmrestd; do
               charmcraft upload "$release" "$charm$extension"
```
Comment (heitorPB) [inline .github/workflows/publish-candidate.yaml]:
```suggestion
              charmcraft upload $release "$charm$extension"
```

File: .github/workflows/publish-candidate.yaml [hunk 2]
```diff
@@ -68,6 +68,7 @@ jobs:
           release="--release=edge --release=candidate"
 
           charmcraft upload $release slurmctld$extension --resource etcd:1
+          charmcraft upload $release slurmd$extension --resource nhc:1
 
           for charm in slurmd slurmdbd slurmrestd; do
```
Comment (heitorPB) [inline .github/workflows/publish-candidate.yaml]:
```suggestion
          for charm in slurmdbd slurmrestd; do
```

File: CHANGELOG [hunk 3]
```diff
@@ -6,7 +6,7 @@ This file keeps track of all notable changes to the Slurm Charms.
 
 Unreleased
 ----------
-
+- add NHC resource to slurmd
```
Comment (heitorPB) [inline CHANGELOG]:
```suggestion
- updated NHC to version 1.4.3
- changed how NHC is installed to be via Juju resources
```

File: charm-slurmd/metadata.yaml [hunk 4]
```diff
@@ -44,4 +44,9 @@ resources:
     type: file
     filename: singularity-ce-3.10.2-1.el7.x86_64.rpm
     description: |
-      Official .rpm package to install Singularity. Retrieved from GitHub Releases.
\ No newline at end of file
+      Official .rpm package to install Singularity. Retrieved from GitHub Releases
```
Comment (heitorPB) [inline charm-slurmd/metadata.yaml]:
```suggestion
      Official .rpm package to install Singularity. Retrieved from GitHub Releases.
```

File: .github/workflows/publish-candidate.yaml [hunk 5]
```diff
@@ -68,7 +68,8 @@ jobs:
           release="--release=edge --release=candidate"
 
           charmcraft upload $release slurmctld$extension --resource etcd:1
+          charmcraft upload $release slurmd$extension --resource nhc:1
 
-          for charm in slurmd slurmdbd slurmrestd; do
-              charmcraft upload "$release" "$charm$extension"
+          for charm in slurmdbd slurmrestd; do
+              charmcraft upload $release "$charm$extension"
```
Comment (heitorPB) [inline .github/workflows/publish-candidate.yaml:74]: Please also apply these changes to the github/workflows/publish-edge.yaml file

Comment (omnivector-kelly) [reply .github/workflows/publish-candidate.yaml:74]: Done in [d351c2b](https://github.com/omnivector-solutions/slurm-charms/pull/151/commits/d351c2b7a57f0b1001924732b018b4abbda39c78)

**Review by heitorPB — APPROVED** [review]
Nice work!

**Pull request merged**